src/test/java/com/google/javascript/cl2dts/lends.js

```diff
@@ -0,0 +1,19 @@
+goog.provide('lends');
+goog.provide('lends.A');
+
+/**
+ * @constructor
+ */
+lends.A = function() {
+  /** @type {string} */
+  this.a = "";
+};
+
+goog.object.extend(lends.A, /** @lends {lends.A} */ {b: 1});
+
+// lends manages to avoid the deduping of fields, and as far as closure
+// is concerned there is a field a on the prototype and on the instance.
+// Also the presense of @lends extends the type signature of A.prototype.
+goog.object.extend(lends.prototype.A, /** @lends {lends.A.prototype} */ {a: 1, c: true});
```

Comment:
is this really `lends.prototype.A` on the LHS, and not `lends.A.prototype`? That seems super confusing to me (but then again, that wouldn't be new...).

Assuming it should be the latter, this looks to me as if `a` should end up on `lends.A` and on `lends.A.prototype`, but maybe Closure syntax is sort of overlapping here, confusing `lends.A` the instance type (vs `lends.A` the static container) and `lends.A.prototype`, which ends up mostly being the instance type?

It might be worth raising this on the mailing list. Your fix still seems reasonable.